Array_Vector_Exercises/Exercise1/Menu.cpp

```diff
@@ -0,0 +1,380 @@
+			}
+
+			else
+			{
+				std::system("cls");
+				std::cout << "These are the shapes that you created: " << std::endl;
+
+				for (int i = 0; i < Shapes.GetSize(); i++)
+				{
+					std::cout << "--------------------------------- " << std::endl;
+					std::cout << "Shape #" << i + 1 << std::endl;
+					PrintArray(*Shapes[i]);
+				}
+			}
+
+			std::cout << "Press 1... to back previus menu" << std::endl;
+			int MenuInput;
+			std::cin >> MenuInput;
+
+			while (std::cin.fail() || MenuInput != 1)
+			{
+				std::cin.clear();
+				std::cin.ignore(100, '\n');
+				std::cout << "Invalid Option, please enter a valid Option " << std::endl;
+				std::cin >> MenuInput;
+				if (MenuInput == 1)
+				{
+					ActiveMenuCommand = EMenuCommand::DynamicArray;
+				}
+			}
+		}
+
+		else
+		{
+			//freeing Memory
+			for (int i = 0; i < Shapes.GetSize(); i++)
+			{
+				delete Shapes[i];
+			}
+			ActiveMenuCommand = EMenuCommand::MainMenu;
+		}
+
+	} while (UserInput != 3);
+}
+
+FShape* FMenu::CreateShape()
+{
+	std::cout << "-------------------------------------" << std::endl;
+	std::cout << "1- Circle" << std::endl;
+	std::cout << "2- Square" << std::endl;
+	std::cout << "-------------------------------------" << std::endl;
+
+	int UserInput;
+	UserInput = ValidateInput(UserInput, 1, 2);
+
+	if (UserInput == 1)
+	{
+		std::cout << "Insert circle's Radius: ";
+		float InRadius;
+		
+		InRadius = ValidateInputFloat(InRadius);
+		return new FCircle(InRadius);
+		
+			
+	}
+	else
+	{
+		std::cout << "Insert square's lenght: ";
+		float InLenght;
+		InLenght = ValidateInputFloat(InLenght);
+
+		std::cout << "Insert square's heignt: ";
+		float InHeight;
+		InHeight = ValidateInputFloat(InHeight);
+
+		return new FSquare(InLenght, InHeight);
+	}
+}
+
+//---------------------------------------------------------------------------------
+void FMenu::PrintArray(const FShape& Shape) const
+{
+	std::cout << "this Shape is a: " << Shape.GetKindOfShape() << std::endl;
+	std::cout << Shape.PrintParameters() << std::endl;
+	std::cout << "the Shape's area is: " << Shape.GetArea() << std::endl;
+	std::cout << "the Shape's perimeter is: " << Shape.GetPerimeter() << std::endl;
+	std::cout << "--------------------------------- " << std::endl;
+};
+
+//---------------------------------------------------------------------------------
+void FMenu::RunApp()
+{
+	while (ActiveMenuCommand != EMenuCommand::Exit)
+	{
+		switch (ActiveMenuCommand)
+		{
+		case EMenuCommand::MainMenu:
+			OpenMainMenu();
+			break;
+		case EMenuCommand::BuildInArray:
+			AskForBuildInArray();
+			break;
+		case EMenuCommand::StaticArray:
+			AskForStaticArray();
+			break;
+		case EMenuCommand::DynamicArray:
+			AskForDynamicArray();
+			break;
+		case EMenuCommand::Exit:
+			break;
+		}
+	}
+}
+
+//------------------------------------------------------------------------------
+int FMenu::ValidateInput(int& UserInput, const int LowerBounce, const int HigherBounce)
```

Comment:
We mentioned this in earlier reviews. Why would you want to have an out parameter AND a return type that both return the same thing? Why not just leave the return type, what's the purpose of the UserInput variable?

Reply:
Also, Bounce is misspelled here, it should be Bound.